curp/src/rpc/connect.rs

```diff
@@ -64,15 +66,16 @@ impl FromTonicChannel for InnerProtocolClient<Channel> {
 /// Connect to a server
 async fn connect_to<Client: FromTonicChannel>(
     id: ServerId,
-    mut addrs: Vec<String>,
+    addrs: Vec<String>,
+    #[cfg(not(madsim))] tls_config: Option<ClientTlsConfig>,
```

Comment:
We can define two `connect_to` methods, one with `#[cfg(not(madsim))]` while the other with `#[cfg(madsim)]`. And define a mock `ClientTlsConfig>` in `simulation/src/lib.rs`. Using conditional compile to pick which one should be use. Therefore, we can remove many `#[cfg(madsim)]` in other codes of this pr.

Reply:
Refactored